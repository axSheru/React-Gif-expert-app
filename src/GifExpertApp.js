import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories );

    /* const handleAdd = () => {
        setCategories( cats => [ ...cats, 'Demon Slayer' ] );
    }; */

    return (
        <>
            <h2>Gif expert app</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category => 
                        <GifGrid
                            key={ category }
                            category={ category }
                        />
                    )
                }
            </ol>
        </>
    );

};

export default GifExpertApp;