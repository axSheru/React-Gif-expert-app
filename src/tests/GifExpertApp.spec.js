import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el componente GifExpertApp.', () => {

    test('debe de hacer match con el snapshot.', () => {

        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorÃ­as.', () => {

        const categories = [ 'The mandalorian', 'Back to the future' ];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );



    });

});