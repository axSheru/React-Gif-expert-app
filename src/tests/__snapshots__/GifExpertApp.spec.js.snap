// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp. debe de hacer match con el snapshot. 1`] = `
<Fragment>
  <h2>
    Gif expert app
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
